### ISSUE TITLE
N-Back Style Adaptive Difficulty

### ISSUE THREAD

**Issue by GiladKingsley** [4skinSkywalker/Syllogimous-v4#8]
# N-Back Style Adaptive Difficulty

### Overview
A system inspired by the classic Dual N-Back where if your accuracy is 90% or above, you level up, and if it's below 70%, you level down.

This system will work separately for each question type and will be one of the two options for the Per-Type Difficulty system, with a toggle available in **Settings**.

---

### Mechanics
- If at least the last 10 rounds in that question type were of the same difficulty:
  - **9 or 10 correct**: Difficulty increases.
  - **7 or 8 correct**: Difficulty remains the same.
  - **6 or fewer correct**: Difficulty decreases.

**Comment by 4skinSkywalker** [comment]
This could replace the current score system. I may want to try it, but what's the benefit of it? Having 4 mistakes in a rows means level down, is it good? Dunno.
Can you provide some reasoning on why this might be a better system?

**Comment by GiladKingsley** [comment]
Having 4 mistakes in the row will not mean immediately leveling down. 

The level down will occur only after 10 trials, in which four of them are wrong, if 4 are wrong consecutively, you will still get play up to 6 more rounds in that level.

The benefit of this, as I see it, is to control for difficulty so that the player is always in a challenging but achievable zone, where learning is maximized.

Additionally, 40% failure rate is quite high given 50% is random guessing, so unless it feels really wrong in testing, I think it's gonna be fine.

**Comment by 4skinSkywalker** [comment]
But what's the point of "you will still get play up to 6 more rounds in that level." if you already know you have regressed? xD I can just level down and that way it's even more reactive to the level of difficulty

**Comment by GiladKingsley** [comment]
I see your point, I agree.
I just defaulted to that because that's most similar to how Dual N-Back works mechanistically; even if you know you have done terribly, I think you still get to play the rest of the round.
Unlike N-Back though, here, getting 4 wrong probably takes longer than a round of Dual N-Back. So it does make sense to immediately level down.
Game-wise too, you're right, it could feel pointless as a player to play your best if you already know you will level down.
Also, getting 4 wrong in a row would be quite rare given you have a 50% chance of guessing each answer correctly.

**Comment by 4skinSkywalker** [comment]
Can I close this one? :)

**Comment by GiladKingsley** [comment]
Should we make it so that reaching the premise increase threshold instantly increases the premise number?
It's like the case you brought up above just the other way, if you already reached that accuracy, why not level up already?

**Comment by 4skinSkywalker** [comment]
Same case for the level down then

**Comment by 4skinSkywalker** [comment]
@GiladKingsley what do we do with this?

**Comment by GiladKingsley** [comment]
Was the immediate difficulty increase/decrease ever added? I think that was the last thing missing

**Comment by 4skinSkywalker** [comment]
Nope, is it needed? I've added more explanation on the premises up/down event

**Comment by GiladKingsley** [comment]
It just feels wrong leveling up when you were just incorrect, or vice versa

**Comment by 4skinSkywalker** [comment]
In any case your proposal doesn't fix it

**Comment by GiladKingsley** [comment]
Why? Isn't the fix - not necessitating the sample size but rather using it as a maximal reference cutoff?

**Comment by 4skinSkywalker** [comment]
It would trigger at percentage so it still happens that you get level up on wrong or level down or right, unless we do it in such a way that a level up is done on the next correct answer and viceversa. But this would be a gimmick

**Comment by 4skinSkywalker** [comment]
Closing this for the time being
